### ISSUE TITLE
Compile failures with GHC 7.8

### ISSUE THREAD

**Issue by hvr** [andrewthad/chronos#2]
I've noticed that all releases of `chronos` declare support for `base >= 4.7`, however GHC disagrees:

```
Configuring component lib from chronos-0.3
Preprocessing library chronos-0.3...
[ 1 of 22] Compiling Chronos.Internal.CTimespec ( src/Chronos/Internal/CTimespec.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Internal/CTimespec.o )
[ 2 of 22] Compiling Chronos.Internal ( src/Chronos/Internal.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Internal.o )
[ 3 of 22] Compiling Chronos.Types    ( src/Chronos/Types.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Types.o )
[ 4 of 22] Compiling Chronos.Calendar ( src/Chronos/Calendar.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Calendar.o )
[ 5 of 22] Compiling Chronos.Match    ( src/Chronos/Match.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Match.o )
[ 6 of 22] Compiling Chronos.Nanoseconds ( src/Chronos/Nanoseconds.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Nanoseconds.o )
[ 7 of 22] Compiling Chronos.Internal.Conversion ( src/Chronos/Internal/Conversion.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Internal/Conversion.o )
[ 8 of 22] Compiling Chronos.Posix    ( src/Chronos/Posix.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Posix.o )
[ 9 of 22] Compiling Chronos.Day      ( src/Chronos/Day.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Day.o )
[10 of 22] Compiling Chronos.Internal.Format ( src/Chronos/Internal/Format.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Internal/Format.o )
[11 of 22] Compiling Chronos.Date.Text ( src/Chronos/Date/Text.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Date/Text.o )
[12 of 22] Compiling Chronos.Date.ByteString.Char7 ( src/Chronos/Date/ByteString/Char7.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Date/ByteString/Char7.o )
[13 of 22] Compiling Chronos.Datetime ( src/Chronos/Datetime.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Datetime.o )
[14 of 22] Compiling Chronos.TimeOfDay.Text ( src/Chronos/TimeOfDay/Text.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/TimeOfDay/Text.o )
[15 of 22] Compiling Chronos.Datetime.Text ( src/Chronos/Datetime/Text.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Datetime/Text.o )
[16 of 22] Compiling Chronos.TimeOfDay.ByteString.Char7 ( src/Chronos/TimeOfDay/ByteString/Char7.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/TimeOfDay/ByteString/Char7.o )
[17 of 22] Compiling Chronos.Datetime.ByteString.Char7 ( src/Chronos/Datetime/ByteString/Char7.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/Datetime/ByteString/Char7.o )
[18 of 22] Compiling Chronos.OffsetDatetime.Text ( src/Chronos/OffsetDatetime/Text.hs, /tmp/matrix-worker/1480432575/dist-newstyle/build/x86_64-linux/ghc-7.8.4/chronos-0.3/build/Chronos/OffsetDatetime/Text.o )

src/Chronos/OffsetDatetime/Text.hs:40:3:
    Not in scope: ‘<$>’
    Perhaps you meant one of these:
      ‘Atto.<?>’ (imported from Data.Attoparsec.Text),
      ‘<>’ (imported from Data.Monoid)

src/Chronos/OffsetDatetime/Text.hs:41:3:
    Not in scope: ‘<*>’
    Perhaps you meant one of these:
      ‘Atto.<?>’ (imported from Data.Attoparsec.Text),
      ‘Atto.<*.’ (imported from Data.Attoparsec.Text),
      ‘Atto..*>’ (imported from Data.Attoparsec.Text)

```

You can correct the meta-data for existing releases on Hackage yourself via

 - https://hackage.haskell.org/package/chronos-0.3/chronos.cabal/edit

or I can do it for you.

While at it, the `description` ("Please see README.md") is not a proper description for your package (and Hackage will show a link to the README at the end of the description *if* the README is contained in the source tarball (however, it is not); therefore that phrase is redundant), and hence this will cause poor experience for users (NB: your package won't be properly indexed by Hackage's package search feature and thus hurt its discoverability). See also [this posting](https://mail.haskell.org/pipermail/haskell-cafe/2016-November/125541.html) for more details.

If you have any questions, please let me know!

**Comment by andrewthad** [comment]
Thanks for bringing this up! I've improved the bounds (of `base` and of other dependencies) and have corrected the description field. I've released this as revision 1 of chronos-0.3.0.

**Comment by hvr** [comment]
Thanks! :-)
